cni/network/network_test.go

```diff
@@ -1246,6 +1251,22 @@ func (n *InterfaceGetterMock) GetNetworkInterfaces() ([]net.Interface, error) {
 	return n.interfaces, nil
 }
 
+func (n *InterfaceGetterMock) GetNetworkInterfaceAddrs(iface *net.Interface) ([]net.Addr, error) {
+	if n.err != nil {
+		return nil, n.err
+	}
+
+	// actual net.Addr invokes syscall; here just create a mocked net.Addr{}
+	res := []net.Addr{}
+	for _, intf := range n.interfaces {
+		if iface.Name == intf.Name {
+			netAddr := n.intfAddr[iface.Name].ipNet
```

Comment:
Could we change the map value type to a []net.Addr (or if following your code's types more closely, []net.IPNet)-- is there a reason we set the type as interfaceAddr-- why can't we just use the single field's type ("ipnet net.IPNet") directly? 

Right now I think we can only return a single result in the slice/array that is returned. We can only map 1 net.Interface (the iface passed in) to 1 of the interfaces in our map anyway right? If so, we can probably just return the value from the if statement itself instead of going through the whole loop.